README.adoc

```diff
@@ -28,15 +28,132 @@ Synposis
+
+        . *roles/ansible-deploy-rudder/meta* Contains metadata used to deploy the role in Ansible Galaxy.
 
-        . *defaults*: Contains variables used in Rudder installation tasks.
+=== Modules
 
-        . *meta* Contains metadata used to deploy the role in Ansible Galaxy.
+This section aims to give and explain the different parameters possible in each module.
+
+==== Module: 'rudder_server_settings'
+
+This module is used to set up Rudder via API calls.
+
+.Table 'rudder_server_settings' module options
+[cols="1,1,1,3"] 
+|===
+|Name | Type |Required|Description
+
+|rudder_url
+|*false*
+|str
+|Providing Rudder server IP address. Defaults to localhost of the target node if not set, with certificate validation disabled, unless explicitly enabled by setting validate_certs.
+
+|rudder_token
+|*false*
+|str
+|Providing Rudder server token. Defaults to the content of /var/rudder/run/api-token if not set.
+
+|name
+|*true*
+|str
+|The name of the parameter to set.
+
+|value
+|*true*
+|str
+|The value defined to modify a given parameter name.
+
+|validate_certs
+|*false*
+|boolean
+|Choosing either to ignore or not Rudder certificate validation. Defaults to *true*.
+|===
+
+Example of use:
+
+----
+     rudder_server_settings:
+       rudder_url: "https://localhost/rudder"
+       name: "enable_self_validation"
+       value: "false"
+       validate_certs: False
+----
+
+==== Module: 'rudder_node_settings'
+
+This module allows you to configure the settings of the nodes in Rudder.
+
+.Table 'rudder_server_settings' module options
+[cols="1,1,1,3"] 
+|===
+|Name | Type |Required|Description
+
+|rudder_url
+|*false*
+|str
+|Providing Rudder server IP address. Defaults to localhost of the target node if not set, with certificate validation disabled, unless explicitly enabled by setting validate_certs.
+
+|rudder_token
+|*false*
+|str
+|Providing Rudder server token. Defaults to the content of /var/rudder/run/api-token if not set.
+
+|validate_certs
+|*false*
+|boolean
+|Choosing either to ignore or not Rudder certificate validation. Defaults to true.
+
+|node_id
+|*true*
+|str
+|Define the identifier of the node to be configured
+
+|policy_mode
+|*false*
+|str
+|Set the policy mode to (default, enforce or audit)
+
+|state
+|*false*
+|str
+|Set the node life cycle state to (enabled, ignored, empty-policies, initializing or preparing-eol)
+
+|properties
+|*false*
+|dict
+|Define a properties (with "name:" and "value:")
+
+|agent_key
+|*false*
+|dict
+|Define information about agent key or certificate (update PEM with "value:<PEM>" and status with "status:<certified|undefined>")
+|===
+
+Example of use:
+
+----
+    rudder_node_settings:
+        rudder_url: "https://localhost/rudder"
+        node_id: root
+        policy_mode: enforce
+        state: enabled
+        validate_certs: False
+        properties:
+          name: env_type
+          value: prod
+        agent_key:
+          status: undefined
```

Comment:
a realy example for the agent key would be great
I tried with 

`value: "-----BEGIN CERTIFICATE-----\nMIIFSzCCAzOgAwIBAgIUQsIEVqZwxFVqLyqGf/KosnzAMJ8wDQYJKoZIhvcNAQEL\nBQAwFjEUMBIGCgmSJomT8ixkAQEMBHJvb3QwHhcNMjEwNTIxMDc1NjU5WhcNMzEw\nNTE5MDc1NjU5WjAWMRQwEgYKCZImiZPyLGQBAQwEcm9vdDCCAiIwDQYJKoZIhvcN\nAQEBBQADggIPADCCAgoCggIBALJ1hTqm1EpN215ZBOYTUyIQXrSv3/IMGgyP0bPZ\nHMX2nhYpO+ydJB1DQ3PEV3V7z/N5gXf+RwRvkGiITLuPN7U/eWjOalpqUFz2rVkr\nyIR0falNHxLu7XDP/zC+tTK4U+uDJZz415EzSgXFNEXRoOOu5my8Cw1I+PcCrPeP\n0lCB4x5Qreqg7GpwQxuys7op0ToxxoBtniQNdMfBR+DfBEFbAlJ/zIdM0+h5qa29\np/kBWTineLwM5L+WwwdWyVP0KP5RNOwPsm/b9p9j48YFLetuSXwqVvJWN5ulOM4f\nlaM4LVnBZIh7ybgty7tazfow7atWiTvw3H5RO8CfNTFDgupCavYxOumkac8Vj82H\nu6Uz40KD01kAA7Yhwzt+CjHt9+a5LcP3fU5XcvdlhC2nBEp2jV9/tyWUGCe7K+OZ\ncQoNaBEeDhV4U6ril6LZL21Kqzsr8krHOvwzgmPI9lpB7k+B+IMwiQ+fRrPslSJ9\n/ybwJ9SuzRkNhblxExT/IsHAyo3e1QM3Tg0pH7t3Nti2LqxTTAd7nYC35VBJZv6K\nt/lYAgPFQXFxngDUimV9RstEMqYbx0nx+Fcwh5I+ZHTiQ7WNBohMtrxMtArCA+d5\nr8h52tMQ1RJZdHz/BCf1phhykPQJtCEaUsOjIuBbj4BcoMx4qP+1zsGPSgkY8zOo\nCLjxAgMBAAGjgZAwgY0wDAYDVR0TBAUwAwEB/zAdBgNVHQ4EFgQU0r6r9MwvgBvk\nsjh0Z/Nc3oV1W58wUQYDVR0jBEowSIAU0r6r9MwvgBvksjh0Z/Nc3oV1W5+hGqQY\nMBYxFDASBgoJkiaJk/IsZAEBDARyb290ghRCwgRWpnDEVWovKoZ/8qiyfMAwnzAL\nBgNVHQ8EBAMCArwwDQYJKoZIhvcNAQELBQADggIBAAnOBQM2qrLiY6RtzW2GgT75\nbnonFQpSZ538qwPHd605ItBmhw044mmjTKhV7jvPSmba6xsPO9syi7Jk7ZzanaWv\nunjxbX+go1GXErtizQGlcMrJkQvEWAieKeL7CgPMgUb0WTe1rd/juuh1Ty1ceiDY\nGgXUgUoxVAp5+n9szvz1wwlwA4l5E0TdnQJTXysCkm1Y5o20ij363nnr2Jhf/z2w\nybWXSh6lak1RKhTyAojzkbQcMy8E273x32QDf9942bQBtff1NOybzquiUKHo1Rke\nXkjwadKjq2rD7MSKkmXIiEYb/Fg/nXLxVRyCpxtOYuwHJOAXtGZjUB8x8C5arPCi\nqWrHEHySKhLMi3b6M176VtjiJmjS61P5HO3SzMZMqaUizkPyfRfR0QDsk8SJbBWU\n3QfABlVsvJTPJqHCmz2C4KLoM4ZMRlx7e4MqZvYlO+dbVvbWTGuxpuSVzwG+w7FD\ntFWthQkGEb56maZQqNgskNgcPaMoFMnjrNcJdN2fhSCBZtLlp555PhOg/ngBjiai\ny8Nck9lhvLnAowoDZ6TPJplfLMH6SjJjlKiRbp786x+RpoqBUf+ONgCIP7DJUmjr\nV0eVtFwSVj/NhM92MsKfeNIvk1bfFWArqnwwVEROoFT6QmjVba9Idq0LDl+Rg7em\nL9XQ0rY+TfnOGyuPHOkk\n-----END CERTIFICATE-----"`
and it failed